pkg/worker/scm/provider/git.go

```diff
@@ -85,11 +85,27 @@ func (g *Git) Clone(token, url, ref, destPath string) (string, error) {
 		args []string
 	}
 	cmds := []cmd{
-		cmd{dir, []string{"clone", url, base}},
+		// --single-branch only checkout the direct branch, it make the process faster.
+		cmd{dir, []string{"clone", "-b", "master", "--single-branch", url, base}},
 		cmd{destPath, []string{"fetch", "origin", ref}},
 		cmd{destPath, []string{"checkout", "-qf", "FETCH_HEAD"}},
 	}
 
+	if strings.HasPrefix(ref, "refs/merge-requests") {
```

Comment:
Add to comments to illustrate that this code is for Gitlab PR.

Reply:
done